### PR TITLE
Add tag extensions

### PR DESCRIPTION
**Pull request by Juuxel**
These extensions allow you to do conditional tag values and tag value replacing (`replace` tag in tag JSONs).

Note: I haven't tested these yet.

### CODE REVIEW

**Comment by Boundarybreaker** [comment]
What would a tag using this look like?

**Comment by Juuxel** [comment]
```json
{
  "replace": false,
  "values": ["vanilla values here"],
  "libcd": {
    "replace": { "optional replace block": "with a condition" },
    "entries": [{
      "when": [{ "libcd:mod_loaded": "lil-tater" }],
      "values": ["lil tater values here"]
    }]
  }
}
```
- libcd.replace only clears existing entries from other datapacks (like vanilla), leaves the vanilla entries in the same json intact
- libcd.replace can also be an array where all conditions must be true
- Each entry (libcd.entries) has its own values and a list of conditions (all must be true so that the values are added)

**Comment by Boundarybreaker** [comment]
That'd be neat.

**Comment by Boundarybreaker** [comment]
@Juuxel Is there anything needed for these other than testing, currently?

File: src/main/java/io/github/cottonmc/libcd/mixin/MixinTagBuilder.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+       and our replacing doesn't clear theirs */
+    @Inject(method = "fromJson", at = @At(value = "INVOKE", target = "Ljava/util/Set;addAll(Ljava/util/Collection;)Z", remap = false))
+    private void onFromJson(Function<Identifier, Optional<T>> getter, JsonObject json, CallbackInfoReturnable<Tag.Builder<T>> info) {
+        try {
+            if (json.has("libcd")) {
+                TagExtensions.ExtensionResult<T> result = TagExtensions.load(
+                        getter,
+                        (blue.endless.jankson.JsonObject) Dynamic.convert(
+                                JsonOps.INSTANCE, JanksonOps.INSTANCE, JsonHelper.getObject(json, "libcd")
+                        )
```
Comment (Juuxel) [inline src/main/java/io/github/cottonmc/libcd/mixin/MixinTagBuilder.java]: This is affected by the issue I fixed in Mojang/DataFixerUpper#42 so we'll need a custom JsonOps instance.

Comment (Juuxel) [reply src/main/java/io/github/cottonmc/libcd/mixin/MixinTagBuilder.java]: Okay, that's now done.

**Comment by Juuxel** [comment]
There shouldn't be anything other than testing left now.

**Comment by Boundarybreaker** [comment]
Alright! I'll give it a try soon.

**Comment by Boundarybreaker** [comment]
This is looking good! Thank you so much!

**Pull request merged**